cmd/cmdexecutor/cmdexecutor.go

```diff
@@ -225,9 +227,18 @@ func getPodNamesUsingLabelSelector(labelSelector, namespace string) ([]types.Nam
 		return nil, err
 	}
 
-	pods, err := core.Instance().GetPods(namespace, selectorsMap)
-	if err != nil {
-		return nil, err
+	var pods *v1.PodList
+
+	for i := 0; i < maxRetries; i++ {
+		pods, err = core.Instance().GetPods(namespace, selectorsMap)
```

Comment:
Does GetPods return error when list of pods is empty ? Should we wait on pod list empty as well ?

Reply:
fixed